rust/src/types/amount.rs

```diff
@@ -370,10 +370,22 @@ impl Shr for u256 {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct U64Overflow;
```

Comment:
Error code could be directly StatusCode. To reduce error casting artifacts. 

If error management turns more complicated, we should consider using a library like thiserror or anyhow instead of polluting the namespace with more symbols. I think we won't be needing neither of these two in the context of the VM .

Reply:
That does not work because in the error case the error gets mapped to different StatusCodes depending on the context.